src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,230 @@
+Passkey authentication Kerberos integration
+=======================================================
+
+Related ticket(s):
+------------------
+ * https://github.com/SSSD/sssd/issues/6228
+
+Problem statement
+-----------------
+This design is an extension of the "enable passkey authentication", where a
+centrally managed user can authenticate locally in a system with a passkey
+token. It describes a Kerberos pre-authentication mechanism using passkey
+authentication. Once it is completed, the received initial ticket granting
+ticket (TGT) can be used for single sign-on into other Kerberos-enabled
+services.
+
+For the purpose of this feature, passkey is a FIDO2 compatible device supported
+by the libfido2 library.
+
+Use cases
+---------
+As a user, not only do I want to walk to any system on a network and log in
+(desktop and/or terminal) using a passkey device connected to the system, but I
+also want to get a Kerberos ticket to identify myself to other services.
+
+Prerequisites
+-------------
+To register a new
+`pre-authentication data identifier <https://www.iana.org/assignments/kerberos-parameters/kerberos-parameters.xhtml#pre-authentication>`__
+in Kerberos for passkey it would be good to first finalize the IdP plugin
+design and security analysis requested by this community.
+
+Solution overview
+-----------------
+The objective is to locally authenticate and issue a Kerberos ticket to a
+centrally managed user. For that purpose SSSD will need to be modified to
+accommodate the new workflow, while still allowing the old one, where the user
+was authenticated locally without issuing any Kerberos ticket. The new workflow
+will be used for online authentication against an IPA server. The old workflow
+will be used for any offline authentication and for AD and other LDAP servers.
+
+SSSD provides a pre-authentication method implementation (pre-auth plugin) to
+MIT Kerberos library. When both the libkrb5 and the KDC support the same
+pre-authentication method, KDC will respond to an AS-REQ message with a request
+to provide an assertion data required by a pre-authentication method. SSSD's
+pre-auth plugin will obtain the assertion data from the passkey. This
+information will be sent by libkrb5 back to the KDC server as a part of the
+AS-REQ exchange. This in turn will relay the request to the ipa-otpd, which
+will retrieve the user information and ask the passkey_child to validate the
+assertion. Once validated the Kerberos KDC will issue a ticket and send it to
+SSSD via libkrb5.
+
+.. image:: ../images/passkey_kerberos_1.png
+  :alt: Architecture diagram
```

Comment:
I'm using plantuml to draw all the diagrams as this is the recommended tool

Reply:
Yes, but mermaid can be made part of the text so it can be easily updated and changes can be tracked.

Reply:
We can also add https://github.com/sphinx-contrib/plantuml (or other extension) to generate the images, I have no problem with that. My goal is to move it from image to text.

Reply:
We do use plantuml in FreeIPA as well. We also embed it in the text.

Reply:
Having the text instead of the image sounds perfect to me. If you can enable some script for plantuml that would be nice, as that's the tool that we should be using.

Reply:
https://github.com/SSSD/sssd.io/pull/66